### PR TITLE
Core/PacketIO: SMSG_DESTRUCTIBLE_BUILDING_DAMAGE

### PR DESCRIPTION
**Pull request by Golrag**
Added SMSG_DESTRUCTIBLE_BUILDING_DAMAGE

Any feedback is welcome ! :)

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -1985,14 +1985,13 @@ void GameObject::ModifyHealth(int32 change, Unit* attackerOrHealer /*= NULL*/, u
...
-        data << player->GetPackGUID();
-        data << uint32(-change);                    // change  < 0 triggers SPELL_BUILDING_HEAL combat log event
-                                                    // change >= 0 triggers SPELL_BUILDING_DAMAGE event
-        data << uint32(spellId);
-        player->SendDirectMessage(&data);
+        WorldPackets::GameObject::DestructibleBuildingDamage packet;
+        packet.Caster = player->GetGUID();
+        packet.Target = GetGUID();
+        packet.Damage = -change;
+        packet.Owner = attackerOrHealer->GetGUID(); // todo: this can be a GameObject
```
Comment (Shauren) [inline src/server/game/Entities/GameObject/GameObject.cpp]: This is strange, don't you think? To me "Owner" is the caster (if unit) or vehicle controller and "Caster" the unit directly dealing damage

Comment (Golrag) [inline src/server/game/Entities/GameObject/GameObject.cpp]: I took the structure from the WPP, and saw 
```
ServerToClient: SMSG_DESTRUCTIBLE_BUILDING_DAMAGE (0x02C8) Length: 45 ConnIdx: 1 Time: 08/03/2015 11:43:27.954 Number: 28452
Target: GameObject Map: 607 (Strand of the Ancients) Entry: 190724 (Gate of the Blue Sapphire)
Caster: Player 
Owner: GameObject Map: 607 (Strand of the Ancients) Entry: 190752 (Massive Seaforium Charge)
Damage: 4500
SpellID: 52408
```

I will swap Caster and Owner :)

Comment (Shauren) [inline src/server/game/Entities/GameObject/GameObject.cpp]: Well, it is possible that fields are swapped in wpp, the order in structures from magic dump isnt absolute - it is only guaranteed to work for that build

Comment (Golrag) [inline src/server/game/Entities/GameObject/GameObject.cpp]: Fixed !

File: src/server/game/Server/Protocol/Opcodes.cpp [hunk 2]
```diff
@@ -1018,7 +1018,7 @@ void OpcodeTable::Initialize()
     DEFINE_SERVER_OPCODE_HANDLER(SMSG_DEFENSE_MESSAGE,                         STATUS_NEVER,        CONNECTION_TYPE_REALM);
     DEFINE_SERVER_OPCODE_HANDLER(SMSG_DELETE_CHAR,                             STATUS_NEVER,        CONNECTION_TYPE_REALM);
     DEFINE_SERVER_OPCODE_HANDLER(SMSG_DESTROY_ARENA_UNIT,                      STATUS_NEVER,        CONNECTION_TYPE_REALM);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_DESTRUCTIBLE_BUILDING_DAMAGE,            STATUS_UNHANDLED,    CONNECTION_TYPE_REALM);
+    DEFINE_SERVER_OPCODE_HANDLER(SMSG_DESTRUCTIBLE_BUILDING_DAMAGE,            STATUS_NEVER,        CONNECTION_TYPE_REALM);
```
Comment (Shauren) [inline src/server/game/Server/Protocol/Opcodes.cpp]: according to the sniff you pasted it should go to CONNECTION_TYPE_INSTANCE

Comment (Golrag) [inline src/server/game/Server/Protocol/Opcodes.cpp]: Updated !

**Comment by tkrokli** [comment]
Nice job, Golrag. :+1: Only constructive criticism as I can see. Keep up the good work! :smiley:
Also: Happy New Year! :fireworks:

**Pull request merged**